common/components/whats-new-banner/whats-new-banner.js

```diff
@@ -0,0 +1,72 @@
+function Banner($module) {
+  this.cacheEls($module)
+}
+
+Banner.prototype = {
+  init: function () {
+    this.renderBanner(this.$module)
+  },
+
+  cacheEls: function ($module) {
+    this.$module = $module
+  },
+
+  renderBanner: function ($element) {
+    const contentfulDate = this.getContnetDate($element)
+    const cookieValue = this.getCookie('banner-dismissed')
+
+    if (contentfulDate.toString() === cookieValue.toString()) {
+      this.removeElement($element)
+    } else {
+      this.dismissBanner(this.$module)
+    }
+  },
+
+  dismissBanner: function ($element) {
+    const button = document.getElementById('dismiss-banner-button')
+
+    button.onclick = e => {
+      e.preventDefault()
+      this.removeElement($element)
+      this.setBannerCookie($element)
+    }
+  },
+
+  removeElement: function (element) {
+    if (element.style.display === 'none') {
+      element.style.display = 'block'
+    } else {
+      element.style.display = 'none'
+    }
+  },
+  getCookie: function (cookieName) {
+    const name = cookieName + '='
+    const decodedCookie = decodeURIComponent(document.cookie)
+    const cookie = decodedCookie.split(';')
+
+    for (let i = 0; i < cookie.length; i++) {
+      let c = cookie[i]
+
+      while (c.charAt(0) === ' ') {
+        c = c.substring(1)
+      }
+
+      if (c.indexOf(name) === 0) {
+        return c.substring(name.length, c.length)
+      }
+    }
+
+    return ''
+  },
```

Comment:
Code looks good, I wonder if there's a library we can use for this to make it more resilient? In the server we use https://www.npmjs.com/package/cookie-parser but I realise it's not so easy to use a library like that on the client-side.

Reply:
I did look around at some libraries to do this. But I have used this exact code snippet in a fair few client side applications and has done well AFAIK, and reads fairly well.